app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt

```diff
@@ -151,4 +167,85 @@ class CreditCardsSettingFragment : PreferenceFragmentCompat() {
 
         isCreditCardsListLoaded = true
     }
+
+    /**
+     * Show a warning to set up a pin/password when the device is not secured. This is only used
+     * when BiometricPrompt is unavailable on the device.
+     */
+    override fun showPinDialogWarning(context: Context) {
+        AlertDialog.Builder(context).apply {
+            setTitle(getString(R.string.credit_cards_warning_dialog_title))
+            setMessage(getString(R.string.credit_cards_warning_dialog_message))
+
+            setNegativeButton(getString(R.string.credit_cards_warning_dialog_later)) { _: DialogInterface, _ ->
+                navigateToCreditCardManagementFragment()
+            }
+
+            setPositiveButton(getString(R.string.credit_cards_warning_dialog_set_up_now)) { it: DialogInterface, _ ->
+                it.dismiss()
+                val intent = Intent(Settings.ACTION_SECURITY_SETTINGS)
+                startActivity(intent)
+            }
+
+            create()
+        }.show().secure(activity)
```

Comment:
I was also expecting some related changes too `incrementShowLoginsSecureWarningCount`. If `showPinDialogWarning` was entirely in `BiometricPromptPreferenceFragment`, I would say we would reuse the same counter, but I still think we should repurpose the logins warning count to be generic that can be shared between the 2 since I don't think we need to have a different counter for doing the same thing.

Reply:
What we're doing with that count is unclear to me. If there is a use for it then that should be a separate issue.

Reply:
Incrementing the warning count allows us to determine whether or not we should `shouldShowSecurityPinWarning` or if we have shown the pin warning a number of times and the user decided to ignore it, we just navigate directly. Look for `shouldShowSecurityPinWarning`.

I am a bit cautious about doing this in a separate issue since we have the time now to just address the feedback. There is also no guarantees that we get to working on followups since it's not in the sprint.

Reply:
I will add this. Thanks :)